### DISCUSSION TITLE
Changing the default tab in a panel-tabset

### DISCUSSION THREAD

**Discussion by kmkramer23** [quarto-dev/quarto-cli#6668] [Feature Requests]
### Description

I was trying to set another tab besides the first tab as the default tab in a panel-tabset when the html document launches.  In Rmarkdown this can be done by adding the {.active} attribute to a tab section but that did not appear to work when rendering in Quarto.  I got around the problem by writing some javascript code and putting it in the include-after-body to switch tabs on load.  

Is there a better way of doing this inside a quarto document?

**Comment by mcanouil** [comment]
I am not sure there is a way in Quarto to change the default tab by using as in rmarkdown the keyword/class `.active`.
Although, I believe it could be useful in some cases, I would still recommend to use the first tab as the default which is logical.

@dragonstyle In the current implementation of the tabset, do you think it could be done or worth adding?

**Reply by merrideewouters** [reply]
Definitely this would be useful to flag an important result. A reader could then look at supporting info in the other tabs if they wish to explore further.

**Reply by neeldhara** [reply]
A somewhat related question: is it possible to link to other tabs in a panel-tabset?

**Reply by mcanouil** [reply]
Unfortunately, I don't think there is a way to customise ids in tabsets.
This being said, the following will work:
````qmd
---
title: "Quarto Playground"
format: html
---

This is a playground for Quarto.

::: {.panel-tabset}

### Tab 1

This is the first tab.

### Tab 2

This is the second tab.

:::

::: {style="height: 20in;"}
:::

Link to tab 1: [Tab 1](#tabset-1-1-tab)
````

@dragonstyle @cscheid is there any way to be able to do `### Tab 1 {#my-tab-1}` in tabset header to have something easier to use as cross-reference?

**Reply by mcanouil** [reply]
Somehow I removed the code block opening line.

> Specifically how did you create the link?

Not sure to understand. As any markdown link.

**Comment by mcanouil** [comment]
I opened an issue for enhancement:
- #7260

**Comment by Sytse45** [comment]
I have a related suggestion for enhancement: could you make it possible to find (in Python) which tab is the active one in a .panel-tabset. 
Or did I overlook something and was this implemented already?
I would like to let the user choose which type of input ( input_slider or input_numeric) to use by selecting the corresponding tab in a tabset. The input values in both input types should be synchronised without provoking an infinite loop. Knowing which input type/tab a new value is coming from could avoid such a loop.

**Reply by boshek** [reply]
I believe that this is what https://github.com/quarto-dev/quarto-cli/issues/7260 is for no?

**Comment by Paulj1989** [comment]
@kmkramer23 (or anyone else that might have suggestions) - Is it possible to share the JavaScript workaround in the meantime? I'm trying to change the default tab shown in a tabset and currently having no luck.
